src/ejabberd_c2s.erl

```diff
@@ -520,11 +520,12 @@ wait_for_feature_before_auth({xmlstreamend, _Name}, StateData) ->
     send_trailer(StateData),
     {stop, normal, StateData};
 wait_for_feature_before_auth({xmlstreamerror, _}, StateData) ->
-    send_element_from_server_jid(StateData, mongoose_xmpp_errors:xml_not_well_formed()),
-    send_trailer(StateData),
-    {stop, normal, StateData};
+    c2s_stream_error(mongoose_xmpp_errors:xml_not_well_formed(), StateData);
 wait_for_feature_before_auth(closed, StateData) ->
-    {stop, normal, StateData}.
+    {stop, normal, StateData};
+wait_for_feature_before_auth(Stream, StateData) ->
```

Comment:
I talked about this with my colleagues, and we believe that this catch-all-return-c2s_stream_error should happen in all the steps of the fsm, and not only on `wait_for_feature_before_auth`. Maybe tests for this could be made as well, do you think you can do that? Should be just mechanical changes, let me know what do you think.